src/test/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/compute/v4_9/CloudStackComputePluginTest.java

```diff
@@ -1,894 +1,1426 @@
+    public void testDoRequestInstanceExceptionCloudRequest() throws FogbowException {
+        // set up
+        CloudStackUser cloudStackUser = CLOUD_STACK_USER;;
+        ComputeOrder computeOrder = Mockito.mock(ComputeOrder.class);
+        GetAllServiceOfferingsResponse.ServiceOffering serviceOffice =
+                Mockito.mock(GetAllServiceOfferingsResponse.ServiceOffering.class);
+        DeployVirtualMachineRequest request = Mockito.mock(DeployVirtualMachineRequest.class);
+        GetAllDiskOfferingsResponse.DiskOffering diskOffering =
+                Mockito.mock(GetAllDiskOfferingsResponse.DiskOffering.class);
 
-        String expectedServiceOfferingsRequestUrl = generateExpectedUrl(endpoint, serviceOfferingsCommand,
-                RESPONSE_KEY, JSON);
-        String expectedDiskOfferingsRequestUrl = generateExpectedUrl(endpoint, diskOfferingsCommand,
-                RESPONSE_KEY, JSON);
+        Mockito.doThrow(new FogbowException()).when(this.plugin)
+                .requestDeployVirtualMachine(Mockito.any(), Mockito.any());
 
-        String fakeServiceOfferingId = "fake-service-offering-id";
-        String fakeDiskOfferingId = "fake-disk-offering-id";
+        // exercise
+        try {
+            this.plugin.doRequestInstance(request, serviceOffice, diskOffering, computeOrder, cloudStackUser);
+            Assert.fail();
+        } catch (Exception e) {
+            // verify
+            Mockito.verify(this.plugin, Mockito.times(TestUtils.NEVER_RUN))
+                    .updateComputeOrder(Mockito.eq(computeOrder), Mockito.eq(serviceOffice), Mockito.eq(diskOffering));
+        }
 
-        Map<String, String> expectedParams = new HashMap<>();
-        expectedParams.put(COMMAND_KEY, computeCommand);
-        expectedParams.put(RESPONSE_KEY, JSON);
-        expectedParams.put(ZONE_ID_KEY, fakeZoneId);
-        expectedParams.put(TEMPLATE_ID_KEY, fakeImageId);
-        expectedParams.put(SERVICE_OFFERING_ID_KEY, fakeServiceOfferingId);
-        expectedParams.put(DISK_OFFERING_ID_KEY, fakeDiskOfferingId);
-        expectedParams.put(USER_DATA_KEY, fakeUserDataString);
-        expectedParams.put(NETWORK_IDS_KEY, fakeNetworkIdsString);
-        CloudStackUrlMatcher urlMatcher = new CloudStackUrlMatcher(expectedParams);
+    }
 
-        String serviceOfferingResponse = getListServiceOfferrings(fakeServiceOfferingId, "fake-service-offering",
-                Integer.parseInt(FAKE_CPU_NUMBER), Integer.parseInt(FAKE_MEMORY), FAKE_TAGS);
-        String diskOfferingResponse = getListDiskOfferrings(fakeDiskOfferingId, Integer.parseInt(FAKE_DISK),true);
-        String computeResponse = getDeployVirtualMachineResponse(FAKE_ID);
+    // Test case: Trying to get all ServiceOfferings in the Cloudstack, but it occurs an error
+    @Test
+    public void testGetServiceOfferingsErrorInCloudstack() throws FogbowException, HttpResponseException {
+        // set up
+        CloudStackUser cloudStackUser = CLOUD_STACK_USER;
```

Comment:
do you think that move the cloud_stack_user to the testUtils or to the CloudStackTestUtils would be a good thing? This can be useful to the other plugins

Reply:
I am doing this in the branch "cloudstack-network-test-refator".